### PR TITLE
feat(backend): Add support for JWTs in OAuth token type

### PR DESCRIPTION
**Pull request by wobsoriano**
## Description

This PR adds JWT verification support for OAuth access tokens (`acceptsToken: 'oauth_token'`). Previously, only tokens starting with `oat_` are verified as oauth tokens at edge.

- Used existing JWT functions while extending `verifyJwt` with a headerType parameter to support OAuth access token JWTs ([RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068))
- Added tests for headerType assertions, and OAuth JWT verifications

Resolves USER-2546

## Checklist

- [x] `pnpm test` runs as expected.
- [x] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Summary by CodeRabbit

* **New Features**
  * Accept OAuth-format JWT access tokens and map their payloads into access-token objects.
  * Recognize multiple OAuth JWT header types.

* **Improvements**
  * Configurable JWT header-type validation (single or multiple allowed types).
  * Broader token detection & routing to JWT verification.
  * Enhanced verification reporting with an additional failure classification plus optional status/action details.

* **Tests**
  * Added fixtures and tests covering OAuth JWT formats, verification flows, and error cases.

* **Chores**
  * Release bump recorded in changeset.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/backend/src/tokens/verify.ts [hunk 1]
```diff
@@ -210,6 +209,105 @@ async function verifyOAuthToken(
...
+          tokenType: TokenType.OAuthToken,
+          errors: [
+            new MachineTokenVerificationError({
+              action: TokenVerificationErrorAction.EnsureClerkJWT,
+              message: `Invalid JWT jti claim ${JSON.stringify(jti)}. Expected a valid OAuth access token ID.`,
+              code: MachineTokenVerificationErrorCode.TokenVerificationFailed,
+            }),
+          ],
+        };
+      }
```
[the remark this answers is not in the document]
Comment (jfoshee) [reply packages/backend/src/tokens/verify.ts]: I disagree w/ coderabbit on this one. The format of the `jti` is not critical to the validation. This is fine for now, and we might want to be more flexible.

File: packages/backend/src/tokens/__tests__/verify.test.ts [hunk 2]
```diff
@@ -2,11 +2,23 @@ import { http, HttpResponse } from 'msw';
...
 
 import type { APIKey, IdPOAuthAccessToken, M2MToken } from '../../api';
-import { mockJwks, mockJwt, mockJwtPayload } from '../../fixtures';
+import { createJwt, mockJwks, mockJwt, mockJwtPayload, mockOAuthAccessTokenJwtPayload } from '../../fixtures';
 import { mockVerificationResults } from '../../fixtures/machine';
+import * as VerifyJwtModule from '../../jwt/verifyJwt';
 import { server, validateHeaders } from '../../mock-server';
 import { verifyMachineAuthToken, verifyToken } from '../verify';
 
+// Moved helper from fixtures to test file as requested
```
Comment (jfoshee) [inline packages/backend/src/tokens/__tests__/verify.test.ts]:
```suggestion
```

File: packages/backend/src/tokens/__tests__/verify.test.ts [hunk 3]
```diff
@@ -313,4 +325,137 @@ describe('tokens.verifyMachineAuthToken(token, options)', () => {
...
+      server.use(
+        http.get(
+          'https://api.clerk.test/v1/jwks',
+          validateHeaders(() => {
+            return HttpResponse.json(mockJwks);
+          }),
+        ),
+      );
+
+      const spy = vi.spyOn(VerifyJwtModule, 'verifyJwt').mockResolvedValueOnce({
```
Comment (jfoshee) [inline packages/backend/src/tokens/__tests__/verify.test.ts]: What's the rationale for mocking the `verifyJwt` function? Wouldn't it be better if this was included in the test? (I understand we have to mock the JWKs response; is that why?)

File: packages/backend/src/tokens/__tests__/verify.test.ts [hunk 4]
```diff
@@ -313,4 +325,137 @@ describe('tokens.verifyMachineAuthToken(token, options)', () => {
...
+      const invalidJwt = 'invalid.jwt.token';
+
+      const result = await verifyMachineAuthToken(invalidJwt, {
+        apiUrl: 'https://api.clerk.test',
+        secretKey: 'a-valid-key',
+      });
+
+      expect(result.errors).toBeDefined();
+    });
+  });
```
Comment (jfoshee) [inline packages/backend/src/tokens/__tests__/verify.test.ts:468]: Can we add tests for `alg: "none"` and expired?

File: packages/backend/src/tokens/verify.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (wobsoriano) [reply packages/backend/src/tokens/verify.ts]: Removed the extra `jti` validation for now 👍🏼

File: packages/backend/src/tokens/__tests__/verify.test.ts (hunk 4 above)
Comment (wobsoriano) [reply packages/backend/src/tokens/__tests__/verify.test.ts:468]: Added! thank you

File: packages/backend/src/tokens/__tests__/verify.test.ts (hunk 3 above)
Comment (wobsoriano) [reply packages/backend/src/tokens/__tests__/verify.test.ts]: I was mocking it because the `createOAuthJwt` helper produces a JWT with a fake signature that won't verify. I will update it with a pre-signed OAuth JWT instead to skip mocking `verifyJwt`!

File: packages/backend/src/tokens/machine.ts [hunk 5]
```diff
@@ -8,6 +9,30 @@ export const API_KEY_PREFIX = 'ak_';
 
 const MACHINE_TOKEN_PREFIXES = [M2M_TOKEN_PREFIX, OAUTH_TOKEN_PREFIX, API_KEY_PREFIX] as const;
 
+export const JwtFormatRegExp = /^[a-zA-Z0-9\-_]+\.[a-zA-Z0-9\-_]+\.[a-zA-Z0-9\-_]+$/;
+
+export function isJwtFormat(token: string): boolean {
+  return JwtFormatRegExp.test(token);
+}
```
[the remark this answers is not in the document]
Comment (wobsoriano) [reply packages/backend/src/tokens/machine.ts:16]: this is internal and the variable/function names are descriptive
